src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -547,8 +659,8 @@ private long fetchLatestCommittedOffsetFromSnowflake() {
   private SnowflakeStreamingIngestChannel openChannelForTable() {
     OpenChannelRequest channelRequest =
         OpenChannelRequest.builder(this.channelName)
-            .setDBName(this.databaseName)
-            .setSchemaName(this.schemaName)
+            .setDBName(this.sfConnectorConfig.getOrDefault(Utils.SF_DATABASE, null))
+            .setSchemaName(this.sfConnectorConfig.getOrDefault(Utils.SF_SCHEMA, null))
```

Comment:
this doesn't seems right, we have to have database and schema, right?

Reply:
Correct but I am relying on the open channel API to send an exception if it passes null.

Reply:
Infact this should never happen and would fail earlier.

Reply:
It's not harmful to keep this but let me remove it. I will just replace it with `map.get()`